### PR TITLE
Download architecture specific laucher

### PR DESCRIPTION
**Pull request by peterdeme**
## Description of the change

Once the backend change is merged, this can be merged as well. Keeping it a draft until then.

Related: https://github.com/spacelift-io/spacelift-worker-image/pull/35
## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue);
- [x] New feature (non-breaking change that adds functionality);
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected);
- [ ] Documentation (a documentation or example fix not affecting the infrastructure managed by this module);

### Development

- [ ] All necessary variables have been defined, with defaults if applicable;
- [ ] The code is formatted properly;

### Code review

- [ ] The module version is bumped accordingly;
- [ ] Spacelift tests are passing;
- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached;
- [ ] This pull request is no longer marked as "draft";
- [ ] Reviewers have been assigned;
- [ ] Changes have been reviewed by at least one other engineer;

### CODE REVIEW

File: asg.tf [hunk 1]
```diff
@@ -9,15 +9,27 @@ set -e
   EOF
 
   user_data_tail = <<EOF
+currentArch=$(uname -m)
```
Comment (peterdeme) [inline asg.tf:12]: fun fact. `arch` variable (function) is [already taken](https://www.geeksforgeeks.org/arch-command-in-linux-with-examples/) in the Unix systems.

I didn't want to use it though, I trust `uname -m` a bit better.

**Review by adamconnelly — COMMENTED** [review]
Looks good minus a small typo. I just haven't marked it as approved yet because I don't think the required backend change has been merged yet.

**Comment by peterdeme** [comment]
@adamconnelly just tested on preprod 🎉 seem to work perfectly fine

image

**Comment by adamconnelly** [comment]
Awesome!

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**